lib/assetmanager.js

```diff
@@ -215,7 +216,37 @@
         var layerIdsWithComponents = [],
             compComponents = [],
             doc = this._document,
-            comps = doc._comps;
+            comps = doc._comps,
+            docMeta = this._document._generatorSettings[META_PLUGIN_ID];
+        
+        if (docMeta && docMeta.json) {
+            docMeta = JSON.parse(docMeta.json);
+        }
+        
+        if (docMeta && docMeta.metaEnabled) {
+            console.log("default components enabled");
+            
+            this._componentManager._defaultComponents = [];
+            //read the default layer spec
+            if (docMeta.scaleSettings) {
+                docMeta.scaleSettings.forEach(function (spec) {
+                    
+                    //make the spec as-expected for component manager
+                    
+                    if (!spec.folder) {
+                        spec.folder = ["/"];
+                    } else if (typeof spec.folder === "string") {
+                        spec.folder = [spec.folder];
+                    }
+                    if (!spec.file) {
+                        spec.file = "";
+                    }
+                    
+                    this._componentManager._defaultComponents.push(spec);
```

Comment:
why are they all defaultComponents? Given the "_" is this something .addMetaComponent(spec) could decide on instead so that part of the implementation stays there?